source/MaterialXGenGlsl/Nodes/UnlitSurfaceNodeGlsl.cpp

```diff
@@ -0,0 +1,56 @@
+//
+// TM & (c) 2022 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenGlsl/Nodes/UnlitSurfaceNodeGlsl.h>
+
+#include <MaterialXGenShader/Shader.h>
+#include <MaterialXGenShader/GenContext.h>
+
+MATERIALX_NAMESPACE_BEGIN
+
+ShaderNodeImplPtr UnlitSurfaceNodeGlsl::create()
+{
+    return std::make_shared<UnlitSurfaceNodeGlsl>();
+}
+
+void UnlitSurfaceNodeGlsl::emitFunctionCall(const ShaderNode& node, GenContext& context, ShaderStage& stage) const
+{
+    const GlslShaderGenerator& shadergen = static_cast<const GlslShaderGenerator&>(context.getShaderGenerator());
+
+    BEGIN_SHADER_STAGE(stage, Stage::PIXEL)
+
+        // Declare the output variable
+        const ShaderOutput* output = node.getOutput();
+        shadergen.emitLineBegin(stage);
+        shadergen.emitOutput(output, true, true, context, stage);
+        shadergen.emitLineEnd(stage);
+
+        const string outColor = output->getVariable() + ".color";
+        const string outTransparency = output->getVariable() + ".transparency";
+
+        const ShaderInput* emission = node.getInput("emission");
+        const ShaderInput* emissionColor = node.getInput("emission_color");
+        shadergen.emitLine(outColor + " = " + shadergen.getUpstreamResult(emission, context) + " * " + shadergen.getUpstreamResult(emissionColor, context), stage);
+        
+        if (context.getOptions().hwTransparency)
```

Comment:
Hi @niklasharrysson, I don't think you want to emit this based on the option as this makes it inconsistent between language implementations. Also, it would make it so you can't change the values which affect "outTransparency" without recompiling the shader.

Reply:
Hey @kwokcb! Those are good points, but this convention has been used for a long time for our surface shading in GLSL., where transparency handling is not generated if the option is not set.

For example, see the PBR version of the surface constructor node here: 
https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXGenGlsl/Nodes/SurfaceNodeGlsl.cpp#L201

And also when emitting the final alpha blending of the fragment result here:
https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXGenGlsl/GlslShaderGenerator.cpp#L643

So shader regeneration has always been required to introduce transparency to a surface shader in GLSL. While I agree it would be nice to improve on this, I think that should be done as part of a more holistic look at our transparency rendering for GLSL. Maybe it's worth adding a separate issue to track this.